tests/trace/test_evaluation_imperative.py

```diff
@@ -279,3 +289,158 @@ def test_evaluation_version_reuse(
         filter=ObjectVersionFilter(base_object_classes=["Evaluation"])
     )
     assert len(evaluations) == 1
+
+
+def generate_evaluation_logger_kwargs_permutations():
+    NOT_SPECIFIED = object()
+
+    class MyModel(weave.Model):
+        const_value: int
+
+        @weave.op
+        def predict(self):
+            return self.const_value
+
+    class MyModelAsync(weave.Model):
+        const_value: int
+
+        @weave.op
+        async def predict(self):
+            await asyncio.sleep(0.01)
+            return self.const_value
+
+    models = [
+        NOT_SPECIFIED,
+        "string_model",
+        {"name": "dict_model"},
+        MyModel(const_value=42),
+        MyModelAsync(const_value=420),
+    ]
+
+    datasets = [
+        NOT_SPECIFIED,
+        "string_dataset",
+        [
+            {"sample": "a", "exp_output": 1},
+            {"sample": "b", "exp_output": 42},
+        ],
+        weave.Dataset(
+            rows=[
+                {"sample": "a", "exp_output": 1},
+                {"sample": "b", "exp_output": 42},
+            ]
+        ),
+    ]
+
+    for model in models:
+        for dataset in datasets:
+            kwargs = {}
+            if model is not NOT_SPECIFIED:
+                kwargs["model"] = model
+            if dataset is not NOT_SPECIFIED:
+                kwargs["dataset"] = dataset
+
+            yield kwargs
+
+
+@pytest.fixture
+def scorer(request):
+    if request.param == "string":
+        return "string_scorer"
+    elif request.param == "dict":
+        return {"name": "dict_scorer"}
+    elif request.param == "weave-scorer":
+
+        class MyScorer(weave.Scorer):
+            @weave.op()
+            def score(self, output: int, exp_output: int):
+                return output == exp_output
+
+        return MyScorer()
+    elif request.param == "weave-scorer-async":
+
+        class MyScorerAsync(weave.Scorer):
+            @weave.op
+            async def score(self, output: int, exp_output: int):
+                await asyncio.sleep(0.01)
+                return output == exp_output
+
+        return MyScorerAsync()
+
+
+@pytest.mark.parametrize(
+    "evaluation_logger_kwargs",
+    generate_evaluation_logger_kwargs_permutations(),
+)
+@pytest.mark.parametrize(
+    "scorer",
+    [
+        "string",
+        "dict",
+        "weave-scorer",
+        "weave-scorer-async",
+    ],
+    indirect=True,
+)
+@pytest.mark.parametrize("score", [0.5, {"value": 0.5, "reason": "woah"}])
```

Comment:
uber-small nit: consider adding a few types of values to the dict (bool is the main one)